Signal/src/ViewControllers/DebugUITableViewController.m

```diff
@@ -67,12 +67,29 @@ + (void)presentDebugUIForThread:(TSThread *)thread
                                                       }],
                                   ]]];
 
-    [contents addSection:[OWSTableSection sectionWithTitle:@"Print to Debug Log"
-                                                     items:@[ [OWSTableItem itemWithTitle:@"Print all sessions"
-                                                                              actionBlock:^{
-                                                                                  [[TSStorageManager sharedManager]
-                                                                                      printAllSessions];
-                                                                              }] ]]];
+    [contents
+        addSection:[OWSTableSection
+                       sectionWithTitle:@"Session State"
+                                  items:@[
+                                      [OWSTableItem itemWithTitle:@"Print all sessions"
+                                                      actionBlock:^{
+                                                          dispatch_async([OWSDispatch sessionStoreQueue], ^{
+                                                              [[TSStorageManager sharedManager] printAllSessions];
```

Comment:
Yeah, I really don't like how unstructured the current solution is.

Some of it's usages are not async though. Here's a good example of a non-trivial consequence of making the sessionStore async:
https://github.com/WhisperSystems/SignalProtocolKit/blob/master/AxolotlKit/Classes/SessionCipher.m#L106

So we'd have to change our encrypt/decrypt to be async/callback based as well.